src/glib/mine/signalproc.cpp

```diff
@@ -102,6 +102,79 @@ void TSum::Save(TSOut& SOut) const {
 }
 
 /////////////////////////////////////////////////
+// Online Summa 
+
+/// Packs sparse vector
+void TSumSpVec::Pack(TIntFltKdV& Data) {
+	TIntFltKdV Tmp;
+	Data.Sort();
+	if (Data.Len() > 0) {
+		Tmp.Add(Data[0]);
+		int Last = Tmp.Last().Key;
+		for (int i = 1; i < Data.Len(); i++) {
+			if (Data[i].Key == Last) {
+				Tmp.Last().Dat += Data[i].Dat;
+			} else {
+				Tmp.Add(Data[i]);
+				Last = Tmp.Last().Key;
+			}
+			if (Tmp.Last().Dat == 0.0) {
+				Tmp.DelLast();
+				if (Tmp.Len() > 0) {
+					Last = Tmp.Last().Key;
+				} else {
+					Last = -1;
+				}
+			}
+		}
+	}
+	Data.Clr();
+	Data.AddV(Tmp);
+}
+
+/// Packs sparse vector
+void TSumSpVec::Update(const TVec<TIntFltKd>& InVal, const uint64& InTmMSecs, const TVec<TIntFltKdV>& OutValV, const TUInt64V& OutTmMSecsV) {
+
```

Comment:
It's better to not re-implement addition of sparse vectors and use:
// z = p * x + q * y
static void TLinAlg::LinComb(const double& p, const TIntFltKdV& x, const double& q, const TIntFltKdV& y, TIntFltKdV& z);